### ISSUE TITLE
False positive: abstract-class-instantiated when base class is not abstract

### ISSUE THREAD

**Issue by hemberger** [PyCQA/pylint#5716]
### Bug description

```python
# pylint: disable=C,R,W
import abc

class Foo:
    @abc.abstractmethod
    def bla(self):
        raise NotImplementedError

class Bar(Foo, metaclass=abc.ABCMeta):
    pass

Bar()
```
According to the `abc.abstractmethod` documentation:
> Using this decorator requires that the class's metaclass is
> ABCMeta or is derived from it.

However, this is not enforced. If the base class is not abstract, then the `@abc.abstractmethod` decorator is simply ignored. While the above code is technically misusing the decorator, it still runs successfully.

The bug in the Pylint logic is that it treats all parent classes as abstract if one of its children is abstract.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:12:0: E0110: Abstract class 'Bar' with abstract methods instantiated (abstract-class-instantiated)
```

### Expected behavior

No error, or a warning about misuse of `@abc.abstractmethod` in non-abstract base class `Foo`.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:41:03) 
[GCC 9.4.0]
```

### OS / Environment

RHEL8

**Comment by Pierre-Sassoulas** [comment]
Closing as I don't think this is a false positive, you can instantiate ``Bar`` but ``bla`` is not implemented so the class is actually abstract.
```python
Traceback (most recent call last):
  File "a.py", line 12, in <module>
    Bar().bla()
  File "a.py", line 7, in bla
    raise NotImplementedError
NotImplementedError
```
